### PR TITLE
Support new default branch behavior

### PR DESCRIPTION
**Pull request by msullivan**
* Default branch is now 'main'
 * --default-branch allows specifying a different default
 * If 'branch' is sent with ClientHandshake, it takes priority.
   If it is `__default__`, then the default branch is used
 * The 'database' ClientHandshake param is used otherwise.
   If it is `edgedb` and `edgedb` does not exist, then
   the default branch is used.
 * Similar behavior for HTTP routes.

### CODE REVIEW

File: edb/server/defines.py [hunk 1]
```diff
@@ -29,7 +29,8 @@
 EDGEDB_SUPERGROUP = 'edgedb_supergroup'
 EDGEDB_SUPERUSER = s_def.EDGEDB_SUPERUSER
 EDGEDB_TEMPLATE_DB = s_def.EDGEDB_TEMPLATE_DB
-EDGEDB_SUPERUSER_DB = 'edgedb'
+EDGEDB_OLD_DEFAULT_DB = 'edgedb'
+EDGEDB_SUPERUSER_DB = 'main'
```
Comment (fantix) [inline edb/server/defines.py:33]: `EDGEDB_SUPERUSER_DB` is also used in 2 functions in `metaschema.py`, should we patch up those 2 functions?

File: edb/server/bootstrap.py [hunk 2]
```diff
@@ -2484,13 +2491,13 @@ async def _bootstrap(ctx: BootstrapContext) -> edbcompiler.CompilerState:
...
-            edbdef.EDGEDB_SUPERUSER_DB,
-            edbdef.EDGEDB_SUPERUSER,
+            default_branch,
+            args.default_database_user or edbdef.EDGEDB_SUPERUSER,
         )
     else:
         await _set_edgedb_database_metadata(
             ctx,
-            edbdef.EDGEDB_SUPERUSER_DB,
+            default_branch,
```
Comment (fantix) [inline edb/server/bootstrap.py:2500]: We should also use `default_branch` in `system_api.py` if not `has_create_database`?

> ```
> async def _ping(tenant):
>     if tenant.get_backend_runtime_params().has_create_database:
>         dbname = edbdef.EDGEDB_SYSTEM_DB
>     else:
>         dbname = edbdef.EDGEDB_SUPERUSER_DB  # FIXME
> ```

**Review by fantix — APPROVED** [review]

File: edb/server/defines.py (hunk 1 above)
Comment (msullivan) [reply edb/server/defines.py:33]: I'm not sure if those functions are even right, really?

Comment (msullivan) [reply edb/server/defines.py:33]: Hmmm... and I bet that patches don't work in single-db mode

Comment (fantix) [reply edb/server/defines.py:33]: Probably true, we don't have a patch test CI on single-db mode. Though the regular single-db CI has been green.

**Comment by fantix** [comment]
Looks like now I can access the same default database with both `edgedb` and `main`:

```
$ edgedb -H localhost --database main query 'select schema::ObjectType { name }'
...
{"name": "default::A"}

$ edgedb -H localhost --database edgedb query 'select schema::ObjectType { name }'
...
{"name": "default::A"}
```

But this doesn't apply on the emulated Postgres endpoint:

```
$ psql -h localhost -U edgedb -p 5656 main -c 'select * from "A"'
 id | __type__
----+----------
(0 rows)

$ psql -h localhost -U edgedb -p 5656 edgedb -c 'select * from "A"'
psql: error: connection to server at "localhost" (::1), port 5656 failed: ERROR:  database 'edgedb' does not exist
```

**Pull request merged**